### PR TITLE
Extends GPUDevicePlugin e2e test to exercise device plugin restarts.

### PR DESCRIPTION
**Pull request by jiayingz**
**What this PR does / why we need it**:
This is part of issue #52189 but does not fix it.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: test/e2e/scheduling/nvidia-gpus.go [hunk 1]
```diff
@@ -240,6 +240,26 @@ var _ = SIGDescribe("[Feature:GPU]", func() {
...
 	f := framework.NewDefaultFramework("device-plugin-gpus")
 	It("run Nvidia GPU Device Plugin tests on Container Optimized OS only", func() {
+		// 1. Verifies GPU resource is successfully advertised on the nodes
+		// and we can run pods using GPUs.
+		By("Starting device plugin daemonset and running GPU pods")
+		testNvidiaGPUsOnCOS(f)
+
+		// 2. Verifies that when the device plugin DaemonSet is removed, resource capacity drops to zero.
+		By("Deleting device plugin daemonset")
+		ds := dsFromManifest(dsYamlUrl)
```
Comment (mindprince) [inline test/e2e/scheduling/nvidia-gpus.go]: Can we move this test to a function?

**Review by rohitagarwal003 — COMMENTED** [review]
Minor nit. LGTM otherwise.

File: test/e2e/scheduling/nvidia-gpus.go [hunk 2]
```diff
@@ -240,6 +240,26 @@ var _ = SIGDescribe("[Feature:GPU]", func() {
...
+		falseVar := false
+		err := f.ClientSet.Extensions().DaemonSets(f.Namespace.Name).Delete(ds.Name, &metav1.DeleteOptions{OrphanDependents: &falseVar})
+		framework.ExpectNoError(err, "failed to delete daemonset")
+		framework.Logf("Successfully deleted device plugin daemonset. Wait for resource to be removed.")
+		// Wait for Nvidia GPUs to be not available on nodes
+		Eventually(func() bool {
+			return !areGPUsAvailableOnAllSchedulableNodes(f)
+		}, 5*time.Minute, time.Second).Should(BeTrue())
```
Comment (mindprince) [inline test/e2e/scheduling/nvidia-gpus.go]: This is failing: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-device-plugin-gpu/20

**Pull request merged**